ElvaRivle/Misc/SidebarData

ElvaRivle opened the pull request:
ElvaRivle/Misc/SidebarData

--------------------------------------------------------------------
 MAIN
 * name and surname saved to redux store on login
 * sidebar shows that name and surname instead of hardcoded data
 * avatar displayed on sidebar instead of hardcoded image
 * Sidebar options now have same icons as Figma
--------------------------------------------------------------------
 OTHER
 * fixed auth redux to accept and save name and surname
 * fixed some older css issues (role font wrong, active link not weight 700)
--------------------------------------------------------------------
 TODO
 * ~~show avatar instead of hardcoded image~~
 * Avatar needs to be in rectangular shape, not circular
 * ~~fix chevron~~ (?)

ElvaRivle committed:
ElvaRivle/Misc/SidebarData

--------------------------------------------------------------------
 MAIN
 * name and surname saved to redux store on login
 * sidebar shows that name and surname instead of hardcoded data
--------------------------------------------------------------------
 OTHER
 * fixed auth redux to accept and save name and surname
--------------------------------------------------------------------
 TODO
 * show avatar instead of hardcoded image

ElvaRivle committed:
ElvaRivle/Misc/SidebarData

--------------------------------------------------------------------
 MAIN
 * name and surname saved to redux store on login
 * sidebar shows that name and surname instead of hardcoded data
 * avatar displayed on sidebar instead of hardcoded image
--------------------------------------------------------------------
 OTHER
 * fixed auth redux to accept and save name and surname
--------------------------------------------------------------------
 TODO
 * ~~show avatar instead of hardcoded image~~

ElvaRivle commented:
![image](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/46258494/38195dc4-ba4b-4b81-95d5-85a31994ba2a)

becarselman commented on src/components/Avatar/Avatar.js:
```diff
@@ -0,0 +1,82 @@
+        return (
+            <Avatar
+                className=""
+                style={{
```
please don't use inline style

becarselman commented on src/components/Avatar/Avatar.js:
```diff
@@ -0,0 +1,82 @@
+        return darkenColor(darkColor, 20);
+    };
+
+    const darkenColor = (color, percent) => {
```
you don't need to do this For avatar we can stick to one color its not problem

becarselman replied:
But if you really want you can leave this, but don't use inline style

becarselman commented on src/components/sidebar/View.js:
```diff
@@ -35,11 +36,12 @@ const Sidebar = ({actions}) => {
             <div className="user-profile-overview">
                 <div className="profile-picture-section">
-                    <img src={SampleProfilePicture} alt="Profile picture"/>
+                    {/*<img src={SampleProfilePicture} alt="Profile picture"/>*/}
```
remove commented code

becarselman commented on src/components/sidebar/Container.js:
```diff
@@ -0,0 +1,21 @@
+    };
+};
+
+const mapDispatchToProps = (dispatch) => ({
```
if you don't have any actions you can delete mapDispatchToProps

becarselman commented on src/sagas/authSaga.js:
```diff
@@ -6,16 +6,22 @@ import {
+    const { token, userId, name, surname } = response;
     localStorage.setItem("accessToken", token);
     yield put(loginSuccess(userId));
+    yield put(setUser( {
```
this will work after login But did you try to refresh page I don't see any call that will get this data back

ElvaRivle replied:
will test once again, but 99% sure that persistStore saves data between refreshes

AntMarkoLovric commented:
Please fix UI as it's in figma, couple of notes: 

- Avatar needs to be in rectangular shape, not circular 
- Name and surname need to be in one line, not multiple 
- Border needs to be thicker
- avatar needs to be centred 
- fix chevron

ElvaRivle commented:
> Avatar needs to be in rectangular shape, not circular

Will fix this when Tarik is completely done with Avatars, then I will create custom rectangular one

> Border needs to be thicker

In figma, border is 1px, same as in code. But I set it to 2px, hopefully it's good enough now

> fix chevron

You mean small icons next to links? I used icons from ant design for those, but if it is necessary, I will use those directly from figma

ElvaRivle committed:
ElvaRivle/Misc/SidebarData

--------------------------------------------------------------------
 MAIN
 * fixed some PR comments, some are TODO
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * ~~show avatar instead of hardcoded image~~
 * Avatar needs to be in rectangular shape, not circular
 * fix chevron

ElvaRivle committed:
ElvaRivle/Misc/SidebarData

--------------------------------------------------------------------
 MAIN
 * Sidebar options now have same icons as Figma
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * ~~show avatar instead of hardcoded image~~
 * Avatar needs to be in rectangular shape, not circular
 * ~~fix chevron~~ (?)

ElvaRivle committed:
ElvaRivle/Misc/SidebarData

--------------------------------------------------------------------
 MAIN
 * Table margins fixed
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * ~~show avatar instead of hardcoded image~~
 * Avatar needs to be in rectangular shape, not circular
 * ~~fix chevron~~ (?)

ElvaRivle committed:
Merge branch 'develop' into ElvaRivle/Misc/SidebarData

# Conflicts:
#	src/components/home/Home.js
